cabal.project-stage0

```diff
@@ -0,0 +1,42 @@
+packages:
+  ./_build/stage0/src/ghc-bin/
+  ./_build/stage0/src/libraries/ghc
+  ./_build/stage0/src/libraries/directory/
+  ./_build/stage0/src/libraries/file-io/
+  ./_build/stage0/src/libraries/filepath/
+  ./_build/stage0/src/libraries/ghc-platform/
+  ./_build/stage0/src/libraries/ghc-boot/
+  ./_build/stage0/src/libraries/ghc-boot-th/
+  ./_build/stage0/src/libraries/ghc-heap
+  ./_build/stage0/src/libraries/ghci
+  ./_build/stage0/src/libraries/os-string/
+  ./_build/stage0/src/libraries/process/
+  ./_build/stage0/src/libraries/semaphore-compat
+  ./_build/stage0/src/libraries/time
+  ./_build/stage0/src/libraries/unix/
+  ./_build/stage0/src/libraries/Win32/
+  ./_build/stage0/src/utils/ghc-pkg
+  ./_build/stage0/src/utils/hsc2hs
+  ./_build/stage0/src/utils/unlit
+  ./_build/stage0/src/utils/genprimopcode/
+  ./_build/stage0/src/utils/deriveConstants/
+
```

Comment:
I don't want to modify the source tree directly, this way it's easier to nuke the `_build` directory and restart from a clean state.

Reply:
I think we don't need to copy all source files across. It is also annoying to do it propely with Make.

cabal will use a separate build directory anyway, only some generated cabal files will end up in the source tree.

Reply:
We need to perform some substitution/file generations more than once (for the different stages). I still think it's cleaner to avoid dirtying the source directory. I don't want to have to fix bugs because we've done something for stage0 and forgot to do it again differently for stage1.